oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/directory/CopyOnReadDirectory.java

```diff
@@ -332,7 +333,7 @@ private void removeDeletedFiles() throws IOException {
 
         Set<String> filesToBeDeleted =
                 // Files present locally
-                ImmutableSet.copyOf(local.listAll()).stream()
+                SetUtils.toLinkedSet(local.listAll()).stream()
```

Comment:
we are simply iterating over `Set`, so there is no need for immutability here.

Reply:
There is also no need to create a null-safe set as we do not do any look-up, so I think we can just remove the call to `ImmutableSet.copyOf()`. The set created by the stream is already immutable, so this call to Guava seems to be redundant.

Reply:
`ImmutableSet.copyOf()` is called to convert the array returned by `local.listAll()` to an collection, and then run stream operations on it.